public/pages/CreateMonitor/containers/DefineMonitor/DefineMonitor.js

```diff
@@ -168,24 +171,38 @@ class DefineMonitor extends Component {
     const { httpClient, values, notifications } = this.props;
     const formikSnapshot = _.cloneDeep(values);
 
-    // If we are running a visual graph query, then we need to run two separate queries
-    // 1. The actual query that will be saved on the monitor, to get accurate query performance stats
-    // 2. The UI generated query that gets [BUCKET_COUNT] times the aggregated buckets to show past history of query
-    // If the query is an extraction query, we can use the same query for results and query performance
-    const searchRequests = [buildSearchRequest(values)];
-    if (values.searchType === SEARCH_TYPE.GRAPH) {
-      searchRequests.push(buildSearchRequest(values, false));
+    const searchType = values.searchType;
+    let requests;
+    switch (searchType) {
+      case SEARCH_TYPE.QUERY:
+        requests = [buildSearchRequest(values)];
+        break;
+      case SEARCH_TYPE.GRAPH:
+        // If we are running a visual graph query, then we need to run two separate queries
+        // 1. The actual query that will be saved on the monitor, to get accurate query performance stats
+        // 2. The UI generated query that gets [BUCKET_COUNT] times the aggregated buckets to show past history of query
+        // If the query is an extraction query, we can use the same query for results and query performance
+        requests = [buildSearchRequest(values)];
+        requests.push(buildSearchRequest(values, false));
+        break;
+      case SEARCH_TYPE.LOCAL_URI:
+        requests = [buildLocalUriRequest(values)];
+        break;
     }
 
     try {
-      const promises = searchRequests.map((searchRequest) => {
+      const promises = requests.map((request) => {
         // Fill in monitor name in case it's empty (in create workflow)
         // Set triggers to empty array so they are not executed (if in edit workflow)
         // Set input search to query/graph query and then use execute API to fill in period_start/period_end
         const monitor = formikToMonitor(values);
         _.set(monitor, 'name', 'TEMP_MONITOR');
         _.set(monitor, 'triggers', []);
-        _.set(monitor, 'inputs[0].search', searchRequest);
+        if (searchType === SEARCH_TYPE.QUERY || searchType === SEARCH_TYPE.GRAPH) {
+          _.set(monitor, 'inputs[0].search', request);
+        } else if (searchType === SEARCH_TYPE.LOCAL_URI) {
+          _.set(monitor, 'inputs[0].uri', request);
+        }
         return httpClient.post('../api/alerting/monitors/_execute', {
```

Comment:
Add an else statement that would log information stating an unsupported `searchType` was found